frontend/src/metabase/visualizations/components/List/VariantInfo.tsx

```diff
@@ -16,20 +18,61 @@ import {
 
 import ListCell from "./ListCell";
 
-export const VariantInfo = ({
+const getNamedColumnIndexes = (listColumnIndexes: ListColumnIndexes) => {
+  const [imageIndex] = listColumnIndexes.image;
+  const [titleIndex, subtitleIndex, subtitle2Index] = listColumnIndexes.left;
+  const [infoIndex] = listColumnIndexes.right;
+
+  return {
+    imageIndex,
+    titleIndex,
+    subtitleIndex,
+    subtitle2Index,
+    infoIndex,
+  };
+};
+
+export const VariantInfoHeader = ({
+  listColumnIndexes,
+  getColumnTitle,
+}: {
+  listColumnIndexes: ListColumnIndexes;
+  getColumnTitle: (columnIndex: number) => string;
+}) => {
+  const { imageIndex, titleIndex, subtitleIndex, subtitle2Index, infoIndex } =
+    getNamedColumnIndexes(listColumnIndexes);
+
+  const leftTitleArray = [
+    getColumnTitle(titleIndex),
+    getColumnTitle(subtitleIndex),
+    getColumnTitle(subtitle2Index),
+  ].filter(Boolean);
+
+  const leftTitleString =
+    leftTitleArray.length === 3
+      ? `${leftTitleArray[0]}, ${leftTitleArray[1]}, and ${leftTitleArray[2]}`
```

Comment:
The "and" here won't get translated. I suppose we'd need something like `ngettext` (like [here](https://github.com/metabase/metabase/blob/84df58aad1aca313d93b18843a5737de18a173da/frontend/src/metabase/components/MetadataInfo/TableInfo/ColumnCount.tsx#L18))